UQL/Interpreter.php

```diff
@@ -158,26 +171,14 @@ public function buildFilterLevel($astSubtree)
 
             return $filterDefinition;
         } elseif ($astSubtree instanceof ASTAssertion) {
-            $dataSourceElement = $this->matchDataSourceElement($astSubtree->getIdentifier());
-            $method = $this->translateOperator($astSubtree->getOperator(), $dataSourceElement);
-            if ($astSubtree->getValue() instanceof ASTFunctionCall) {
-                $value = $this->callFunction($astSubtree->getValue());
-            } else {
-                if ($astSubtree->getOperator() == 'T_OP_IN') {
-                    // We expect an array here
-                    if (!($astSubtree->getValue() instanceof ASTArray)) {
-                        throw new UQLInterpreterException("Only arrays are valid arguments for IN statements");
-                    }
-                    $value = $this->parseArray($astSubtree->getValue()->getElements());
-                } else {
-                    $value = $this->parseValue($astSubtree->getValue());
-                }
-            }
+            $field = $this->matchDataSourceElement($astSubtree->getIdentifier());
+            $method = $this->translateOperator($astSubtree->getOperator(), $field);
+            $value = $this->getValue($astSubtree);
 
-            return new FilterCondition($dataSourceElement->getUniqueName(), $method, $value);
-        } else {
-            throw new UQLInterpreterException('Unexpected Abstract Syntax Tree element');
+            return $this->filterConditionFactory->create($value, $method, $field);
```

Comment:
Any reason why the order of parameters has changed?
Maybe it would less confusing to have the same order of parameters for `FilterConditionFactory::create()` as for `FilterCondition` object constructor?